dask/core.py

```diff
@@ -491,18 +491,19 @@ def inline(dsk, keys=None):
     {'y': (inc, 1), 'z': (add, 1, 'y')}
 
     >>> inline(d, keys='y')  # doctest: +SKIP
+    {'z': (add, 1, (inc, 1))}
+
+    >>> inline(d, keys='y', inline_constants=False)  # doctest: +SKIP
     {'x': 1, 'z': (add, 'x', (inc, 'x'))}
     """
-    # inline all constants if keys is unspecified
     if keys is None:
-        keys = set()
-        for key, val in dsk.items():
-            if not istask(val):
-                keys.add(key)
-    elif not isinstance(keys, list):
-        keys = set([keys])
-    else:
+        keys  = set()
+    elif isinstance(keys, list):
         keys = set(keys)
```

Comment:
Yeah, I know `get` supports it, but I don't know of any other functions that do.  After a quick peruse, it appears the optimization functions are the only ones that this could apply.  Your question makes more sense now.  It wouldn't hurt to flatten lists.